File: src/backend/schemas/simulation.py
"""
模拟送达 Pydantic Schema 定义
"""
from pydantic import BaseModel, Field
from typing import List, Optional


class DeliverRequest(BaseModel):
    """模拟送达请求体"""
    vehicle_code: Optional[str] = Field(None, description="车辆编号（可选）")
    package_code: Optional[str] = Field(None, description="包裹编号（可选）")


class DeliverResponse(BaseModel):
    """模拟送达响应数据"""
    delivered_package_codes: List[str] = Field(..., description="送达包裹编号列表")
    status_changed_goods_count: int = Field(..., description="状态变更的货物数量")
    updated_order_count: int = Field(..., description="更新的订单数量")
    delivered_order_codes: List[str] = Field(..., description="已送达订单编号列表")

File: src/frontend/src/types/simulation.ts
export interface SimulationDeliverPayload {
  /** Mock 联调辅助；真实 API 契约不含此字段 */
  batch_code?: string
  vehicle_code?: string
  package_code?: string
}

/** 后端 DeliverResponse（api-contract-phase6 §3.2） */
export interface SimulationDeliverResponse {
  delivered_package_codes: string[]
  status_changed_goods_count: number
  updated_order_count: number
  delivered_order_codes: string[]
  level_info?: {
    l0_to_l1?: number
    l1_to_l2?: number
  }
}

/** UI 层统一结果 */
export interface SimulationDeliverResult {
  packages_delivered: number
  delivered_package_codes: string[]
  message?: string
}
